### PR TITLE
PLAT-132837: Enhance touch support

### PR DESCRIPTION
**Pull request by 0x64**
Enact-DCO-1.0-Signed-off-by: Seungcheon Baek (sc.baek@lge.com)

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
We want to provide better UX on touchable display.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Tweaks spotlight effect not to show on touch inputs.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
PLAT-132837

### CODE REVIEW

File: TabLayout/TabLayout.js [hunk 1]
```diff
@@ -27,6 +28,13 @@ import Tab from './Tab';
 
 import componentCss from './TabLayout.module.less';
 
+const TouchableCell = Touchable(Cell);
+
+const isTouchMode = () => {
+	const rootContainer = document.querySelector('#root > div');
```
Comment (MikyungKim) [inline TabLayout/TabLayout.js:34]: Please revert this fix and make another ticket for it.

File: samples/sampler/stories/qa/PopupTabLayout.js [hunk 2]
```diff
@@ -103,3 +103,32 @@ export const WithInput = () => {
...
+							<Button>Second Tab</Button>
+						</TabPanel>
+					</TabPanels>
+				</Tab>
+			</PopupTabLayout>
+		</Panel>
+	);
+};
+
+WithoutIcon.storyName = 'without icon';
```
Comment (MikyungKim) [inline samples/sampler/stories/qa/PopupTabLayout.js:134]: Please add a new line.

File: ThemeDecorator/ThemeDecorator.js [hunk 3]
```diff
@@ -196,7 +202,15 @@ const ThemeDecorator = hoc(defaultConfig, (config, Wrapped) => {
 			)
 		);
 	}
-	if (spotlight) App = SpotlightRootDecorator({noAutoFocus}, App);
+	if (spotlight) {
+		App = SpotlightRootDecorator({
+			getConfigEffect: (fn) => {
+				spotlightRootDecoratorController.configEffect = fn;
```
Comment (MikyungKim) [inline ThemeDecorator/ThemeDecorator.js:208]: setInputType

File: ThemeDecorator/ThemeDecorator.js [hunk 4]
```diff
@@ -8,12 +8,14 @@
...
 import {addAll} from '@enact/core/keymap';
 import hoc from '@enact/core/hoc';
+import platform from '@enact/core/platform';
 import I18nDecorator from '@enact/i18n/I18nDecorator';
 import {Component} from 'react';
 import classNames from 'classnames';
 import {ResolutionDecorator} from '@enact/ui/resolution';
 import {FloatingLayerDecorator} from '@enact/ui/FloatingLayer';
 import SpotlightRootDecorator from '@enact/spotlight/SpotlightRootDecorator';
+import LS2Request from '@enact/webos/LS2Request/LS2Request';
```
Comment (MikyungKim) [inline ThemeDecorator/ThemeDecorator.js:18]:
```suggestion
import LS2Request from '@enact/webos/LS2Request';
```

File: ThemeDecorator/ThemeDecorator.js [hunk 5]
```diff
@@ -175,6 +177,8 @@ const ThemeDecorator = hoc(defaultConfig, (config, Wrapped) => {
 		[css.bg]: !overlay
 	});
 
+	let spotlightInputType = {};
```
Comment (MikyungKim) [inline ThemeDecorator/ThemeDecorator.js:180]:
```suggestion
	const spotlightInputType = {};
```

**Review by MikyungKim — APPROVED** [review]
LGTM for now,

@stanca-pop-lgp , would you please make a ticket for fixing below issues?
In other skins, the layou is broken with two buttons and close button.
image

File: useScroll/ScrollbarTrack.module.less [hunk 6]
```diff
@@ -44,7 +44,11 @@
 				border-right: @sand-scrollbar-thumb-focus-direction-indicator-width/2 solid transparent;
 				border-left: @sand-scrollbar-thumb-focus-direction-indicator-width/2 solid transparent;
 			}
-		})
+		});
+
+		:global(.spotlight-input-touch) & {
```
Comment (jeonghee27) [inline useScroll/ScrollbarTrack.module.less:49]: https://github.com/enactjs/enact/pull/2914

File: Slider/Slider.module.less [hunk 7]
```diff
@@ -210,6 +210,27 @@
 			});
 		});
 
+		:global(.spotlight-input-touch) &.pressed {
```
Comment (jeonghee27) [inline Slider/Slider.module.less:213]: https://github.com/enactjs/sandstone/pull/791

File: MediaPlayer/MediaSlider.module.less [hunk 8]
```diff
@@ -109,6 +109,22 @@
 				});
 			});
 
+			:global(.spotlight-input-touch) &.pressed {
```
Comment (jeonghee27) [inline MediaPlayer/MediaSlider.module.less:112]: https://github.com/enactjs/sandstone/pull/836

File: TabLayout/TabLayout.js [hunk 9]
```diff
@@ -273,6 +281,24 @@ const TabLayoutBase = kind({
 				forward('onTabAnimationEnd')
 			)
 		),
+		handleFlick: ({direction, velocityX}, {collapsed, onCollapse, onExpand}) => {
```
Comment (jeonghee27) [inline TabLayout/TabLayout.js:284]: https://github.com/enactjs/sandstone/pull/787

File: TabLayout/TabLayout.js [hunk 10]
```diff
@@ -273,6 +281,24 @@ const TabLayoutBase = kind({
...
+			// See the global class 'spotlight-input-touch' to check the input type is touch
+			if (isTouchMode() && direction === 'horizontal') {
+				if (!collapsed && velocityX < 0) {
+					onCollapse();
+				} else if (collapsed && velocityX > 0) {
+					onExpand();
+				}
+			}
+		},
+		handleClick: handle(
```
Comment (jeonghee27) [inline TabLayout/TabLayout.js:294]: What is the source of this code? If this is an additional implementation, please leave a description

File: CHANGELOG.md [hunk 11]
```diff
@@ -12,6 +12,7 @@ The following is a curated list of changes in the Enact sandstone module, newest
 
 - `sandstone/PopupTabLayout` to collapse its tab only when a user enters a menu
 - `sandstone/Scroller` focus rule to match latest UX when `focusableScrollbar` prop is `byEnter`
+- `sandstone/Slider` to show active style when touch dragging
```
Comment (jeonghee27) [inline CHANGELOG.md:15]: I think we don't need this changelog.
instead we need changelog for https://github.com/enactjs/enact/pull/2869 and general touch support

File: ThemeDecorator/ThemeDecorator.module.less [hunk 12]
```diff
@@ -52,8 +54,6 @@
 	:focus {
 		// Prevent Chrome's default focus treatment
 		outline: none;
-		// Prevent Android Chrome from highlighting tapped elements
-		-webkit-tap-highlight-color: transparent;
```
Comment (jeonghee27) [inline ThemeDecorator/ThemeDecorator.module.less:56]: why this change is different with https://github.com/enactjs/sandstone/pull/784??

File: TabLayout/TabLayout.js (hunk 10 above)
Comment (jeonghee27) [reply TabLayout/TabLayout.js:294]: I found it from https://github.com/enactjs/sandstone/pull/828.
In the comments, PLAT-132782 is omitted.

File: useScroll/useOverscrollEffect.js [hunk 13]
```diff
@@ -4,7 +4,7 @@ import {constants} from '@enact/ui/useScroll';
 import {useCallback, useEffect, useRef} from 'react';
 
 const
-	{overscrollTypeDone, overscrollTypeNone, overscrollTypeOnce} = constants,
+	{overscrollTypeDone, overscrollTypeHold, overscrollTypeNone, overscrollTypeOnce} = constants,
```
Comment (jeonghee27) [inline useScroll/useOverscrollEffect.js:7]: https://github.com/enactjs/sandstone/pull/792

File: TabLayout/TabLayout.js (hunk 10 above)
Comment (0x64) [reply TabLayout/TabLayout.js:294]: #828

File: CHANGELOG.md (hunk 11 above)
Comment (0x64) [reply CHANGELOG.md]: I updated as we discussed.

File: ThemeDecorator/ThemeDecorator.module.less (hunk 12 above)
Comment (0x64) [reply ThemeDecorator/ThemeDecorator.module.less:56]: It was missing in #784.
